aiida_quantumespresso/calculations/pp.py

```diff
@@ -1,27 +1,218 @@
 # -*- coding: utf-8 -*-
-"""`CalcJob` implementation for the pp.x code of Quantum ESPRESSO."""
+""" CalcJob implementation for pp.x """
 from __future__ import absolute_import
 
-from aiida.orm import RemoteData, FolderData
-from aiida_quantumespresso.calculations.namelists import NamelistsCalculation
+import os
+import six
+from six.moves import range
 
+from aiida import orm
+from aiida.common import datastructures, exceptions
 
-class PpCalculation(NamelistsCalculation):
+from aiida_quantumespresso.calculations import _lowercase_dict, _uppercase_dict
+from aiida_quantumespresso.utils.convert import convert_input_to_namelist_entry
+from .base import CalcJob
+
+
+def validate_parameters(value, ctx):  # pylint: disable=unused-argument
+    """ Validate 'parameters' dict."""
+
+    # Check for essential keys
+    try:
+        plot_num = value['INPUTPP']['plot_num']
+    except KeyError:
+        return "parameter '[INPUTPP][plot_num]' must be explicitly set"
+
+    try:
+        dimension = value['PLOT']['iflag']
+    except KeyError:
+        raise exceptions.InputValidationError("'[PLOT][iflag]' must be explicitly set")
+
+    # Check that a valid plot type is requested
+    if plot_num in range(23) and plot_num not in [14, 15, 16]:  # Must be integer in range 0-22, but not 14-16:
+        value['INPUTPP']['plot_num'] = int(plot_num)  # If this test passes, we can safely cast to int
+    else:
+        raise exceptions.InputValidationError("'plot_num' must be an integer in the range 0-23")
+
+    # Check for valid plot dimension:
+    if dimension in range(5):  # Must be in range 0-4:
+        value['PLOT']['iflag'] = int(dimension)
+    else:
+        raise exceptions.InputValidationError("'iflag' must be an integer in the range 0-4")
```

Comment:
you shouldn't raise here but just return the message

Reply:
Grand. Fixed.

Reply:
Sorry that I wasn't more clear, but it should return just a string not an exception instance